embedded-hal-async/src/spi.rs

```diff
@@ -35,13 +35,58 @@ where
     Word: Copy + 'static,
 = impl Future<Output = Result<(), T::Error>>;
 
+#[macro_export]
+/// This macro is a workaround for the workaround in [`SpiDevice::transaction`]
+///
+/// # Examples
+///
+/// ```
+/// use embedded_hal_async::spi::{transaction_helper, SpiBus, SpiBusRead, SpiBusWrite, SpiDevice};
+///
+/// pub async fn transaction_example<SPI>(mut device: SPI) -> Result<u32, SPI::Error>
+/// where
+///     SPI: SpiDevice,
+///     SPI::Bus: SpiBus,
+/// {
+///     transaction_helper!(&mut device, move |bus| async move {
+///         // Unlike `SpiDevice::transaction`, we don't need to
+///         // manually dereference a pointer in order to use the bus.
+///         bus.write(&[42]).await?;
+///         let mut data = [0; 4];
+///         bus.read(&mut data).await?;
+///         Ok(u32::from_be_bytes(data))
+///     })
+///     .await
+/// }
+/// ```
+macro_rules! spi_transaction_helper {
+    ($device:expr, move |$bus:ident| async move $closure_body:expr) => {
+        $crate::spi::SpiDevice::transaction($device, move |$bus| async move {
```

Comment:
I wouldn't force `move` here, you might want to read to a borrowed buffer captured from the parent function.

Reply:
Should all of these combinations be supported?

-  `|bus| async {}`
-  `|bus| async move {}`
-  `move |bus| async {}`
-  `move |bus| async move {}`

Reply:
I *think* just these would be enough

```
|bus| async {}
|bus| async move {}
```

because if the inner `async{}` captures by move the net result is the whole "thing" captures by move, so there's no point in making the outer closure `move` as well. but I'm not sure :sweat:

Reply:
The `transfer` convenience method only works with `move |bus| async move {}`

Reply:
I'm still not sure if we need all combinations, but since the macro supports them all now I'm going to mark this as resolved. Shouldn't hurt to support them all.